src/core/load-fragment.c

```diff
@@ -3509,6 +3509,73 @@ int config_parse_restrict_namespaces(
 }
 #endif
 
+int config_parse_restrict_filesystems(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        _cleanup_strv_free_ char **n = NULL;
+        size_t nlen = 0, nbufsize = 0;
+        char*** restrict_fs = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                /* Empty assignment resets the list */
+                *restrict_fs = strv_free(*restrict_fs);
+                return 0;
+        }
+
+        for (const char *p = rvalue;;) {
+                _cleanup_free_ char *word = NULL, *k = NULL;
+                uint32_t magic_number;
+
+                r = extract_first_word(&p, &word, NULL, EXTRACT_UNQUOTE);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
+                                   "Trailing garbage in %s, ignoring: %s", lvalue, rvalue);
+                        break;
+                }
+
+                k = TAKE_PTR(word);
+
+                r = fs_type_from_string(k, &magic_number);
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                   "Invalid filesystem name '%s', ignoring.", k);
+                        continue;
+                }
+
+                if (!GREEDY_REALLOC(n, nbufsize, nlen + 2))
+                        return log_oom();
+
+                n[nlen++] = TAKE_PTR(k);
+                n[nlen] = NULL;
+        }
+
+        if (n) {
+                r = strv_extend_strv(restrict_fs, n, true);
+                if (r < 0)
+                        return r;
```

Comment:
doesn't matter much, but maybe call strv_sort() here too. just in case, so that the introspected result alwas looks the saame.

Reply:
@iaguis this comment is still unaddressed

Reply:
Hm, this is now a Set which is not ordered. I'm not sure how to handle this. We sort in property_get_restrict_filesystems(), anything else we should do?

Reply:
oh sorry, I had missed that - you can ignore this then